src/lib.rs

```diff
@@ -288,4 +285,40 @@ mod tests {
             Some(&"https://hydra.iohk.io https://iohk.cachix.org https://cache.nixos.org/".into())
         );
     }
+
+    #[test]
+    fn returns_the_same_order() {
+        let res = NixConfig::parse_string(
+            r#"
+                cores = 32
+                experimental-features = flakes nix-command
+                max-jobs = 16
+            "#
+            .into(),
+            None,
+        );
+
+        assert!(res.is_ok());
+
+        let map = res.unwrap();
+
+        // Ensure it's not just luck that it's the same order...
```

Comment:
I wonder if there's a better way to test for this... Right now, just transplanting this test into the current HashMap implementation, I get this to pass like 4/10 attempts...

Reply:
Short of asserting the type is a hashmap I'm not sure how we'd do that. There isn't like a `OrderedIterator` marker or anything.